packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -430,3 +463,153 @@ class _EditableLineWidget extends LeafRenderObjectWidget {
+    endpoints = config.renderObject.getEndpointsForSelection(selection);
+  }
+
+  @override
+  void didUpdateConfig(_TextSelectionHandleOverlay oldConfig) {
+    setState(() {
+      selection = config.selection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
+    });
+  }
+
+  Point _dragPosition;
+  void _dragStart(Point position, _TextSelectionHandleSide side) {
+    _dragPosition = position;
+  }
+
+  void _dragUpdate(double delta, _TextSelectionHandleSide side) {
+    _dragPosition = _dragPosition + new Offset(delta, 0.0);
+    TextPosition position = config.renderObject.getPositionForPoint(_dragPosition);
+
+    TextSelection newSelection;
+    if (side == _TextSelectionHandleSide.left) {
+      newSelection = new TextSelection(
+        baseOffset: position.offset,
+        extentOffset: selection.extentOffset
+      );
+    } else {
+      newSelection = new TextSelection(
+        baseOffset: selection.baseOffset,
+        extentOffset: position.offset
+      );
+    }
+
+    if (newSelection.baseOffset >= newSelection.extentOffset)
+      return;
+
+    setState(() {
+      selection = newSelection;
+      endpoints = config.renderObject.getEndpointsForSelection(selection);
+      config.onSelectionHandleChanged(selection);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Stack(
+      children: <Widget>[
+        new GestureDetector(
+          onHorizontalDragStart: (Point p) { _dragStart(p, _TextSelectionHandleSide.left); },
+          onHorizontalDragUpdate: (double d) { _dragUpdate(d, _TextSelectionHandleSide.left); },
+          child: new CustomPaint(
+            foregroundPainter: new _TextSelectionHandlePainter(
+              side: _TextSelectionHandleSide.left,
+              position: endpoints[0]
+            )
+          )
+        ),
+        new GestureDetector(
+          onHorizontalDragStart: (Point p) { _dragStart(p, _TextSelectionHandleSide.right); },
+          onHorizontalDragUpdate: (double d) { _dragUpdate(d, _TextSelectionHandleSide.right); },
+          child: new CustomPaint(
+            foregroundPainter: new _TextSelectionHandlePainter(
+              side: _TextSelectionHandleSide.right,
+              position: endpoints[1]
+            )
+          )
+        )
+      ]
+    );
+  }
+}
+
+/// Draws a single text selection handle at a fixed screen position.
+class _TextSelectionHandlePainter extends CustomPainter {
+  static const double _kSelectionHandleRadius = 10.0; // pixels
+
+  _TextSelectionHandlePainter({this.side, this.position});
+
+  final Point position;
+  final _TextSelectionHandleSide side;
+
+  // Returns a rect of the given size at the corner of the selection range.
+  Rect _getCornerRect(double size) {
+    if (side == _TextSelectionHandleSide.left)
+      return new Rect.fromLTWH(position.x - size, position.y, size, size);
+    else
+      return new Rect.fromLTWH(position.x, position.y, size, size);
```

Comment:
Here's where you need to know whether the directionality of the selection handle.  It's very possible to have two handles pointed in the same direction if you combine a selection that starts in LTR text and ends in RTL text.